### ISSUE TITLE
Test Failure: avmplus e4x Expressions suite

### ISSUE THREAD

**Issue by Dinnerbone** [ruffle-rs/ruffle#12351]
## Testsuite `from_avmplus/e4x/Expressions/kXMLBadQNameErr`
| Test | Passes? | Expected | Actual |
| ---- | ---- | ---- | ---- |
| Asserting for TypeError | :heavy_check_mark: | | |
| <:x>hi</x> | :x: | `Error #1084` | `Error #1083` |
| <x :ns='attr'>hi</x> | :x: | `Error #1084` | `Error #1083` |

**Comment by Dinnerbone** [comment]
The names roughly correspond to the E4X specification. e11_5_1 states:
> E4X extends the abstract equality comparison algorithm defined by ECMAScript to enable equality comparisons
involving QName and Namespace objects and the types XML and XMLList.

**Comment by Lord-McSweeney** [comment]
The following tests are now passing:
- Expressions/e11_1_1
- Expressions/e11_1_3

**Comment by Lord-McSweeney** [comment]
The following tests are now passing:
- `Expressions/e11_6_1`
- `Expressions/e11_6_2`
- `Expressions/e11_6_3`

**Comment by Lord-McSweeney** [comment]
The only test still failing on the suite `Expressions/e11_3_1` is `Delete an XMLList`.

**Comment by Lord-McSweeney** [comment]
The following tests are now passing:
- `Expressions/e11_1_4`

**Comment by torokati44** [comment]
`Expressions/e11_2_2` got fixed by https://github.com/ruffle-rs/ruffle/pull/13954.

**Comment by sleepycatcoding** [comment]
`Expressions/e11_2_1` got fixed by #14720.

**Comment by Lord-McSweeney** [comment]
When comparing `QName`s' namespaces, avmplus first converts to a string then does a simple string-compare of the namespace lists of both `QName`s, using the below algorithm:
```
    Atom QNameObject::getURI() const
    {
        if (m_mn.isAnyNamespace())
        {
            return nullStringAtom;
        }
        else if (m_mn.namespaceCount() > 1)
        {
            return core()->kEmptyString->atom();
        }
        else
        {
            return m_mn.getNamespace()->getURI()->atom();
        }
    }
```

**Comment by Lord-McSweeney** [comment]
`e11_5_1` was fixed by #16558.

**Comment by evilpie** [comment]
`from_avmplus/e4x/Expressions/e11_3_1` was fixed by #17640.

**Comment by evilpie** [comment]
`from_avmplus/e4x/Expressions/e11_1_2` was fixed by #17711.

**Comment by evilpie** [comment]
Last failure fixed by #17710.
